### PR TITLE
Add asam antora extensions

### PR DESCRIPTION
**Pull request by martinfiebig**
## Describe your changes

Added the submodule 'asam-antora-extensions' from [here](https://code.asam.net/common/asam-antora-extensions). This submodule enables enhanced functionality, i.e. for numbered headings and graphical representation of structure models.

This pull request should fix the issues #66 and #77.

Please note, that in order to ensure the correct generation of numbered headings, the navigation-file '.antora\modules\specification\nav-openmaterial.adoc' is deprecated and the file 'content\nav-openmaterial.adoc' should be used instead. (The old file has to be kept though, as the asam-antora-extensions rely on this file and overwrite it in the generation process).

Also, in the files 'geometry/geometry-index.adoc' and 'material/material-index.adoc', the include:: macros have been replaced by xref: macros. This is necessary to avoid duplicates in the headings and to resemble the OpenSCENARIO way of work (i.e. [here](https://www.asam.net/static_downloads/public/asam-openscenario/2.0.0/language-reference/language-reference.html)).

## Issue ticket number and link

This pull request should resolve the issues #66 and #77.

## Checklist before requesting a review

- [x] I have performed a self-review of my code/documentation.
- [x] My changes generate no new warnings during the documentation generation.

### CODE REVIEW

File: .github/workflows/antora-build.yml [hunk 1]
```diff
@@ -17,16 +17,60 @@ permissions:
...
+      with:
+         path: repo
+         fetch-depth: 0 # test if depth in fetch history is required
+         submodules: recursive
+
+    - name: Checkout documentation
+      if: ${{ github.event_name == 'repository_dispatch' && github.event.client_payload.head_ref }}
+      uses: actions/checkout@v4
+      with:
+        repository: martinfiebig/OpenMATERIAL
```
Comment (ClemensLinnhoff) [inline .github/workflows/antora-build.yml:49]: Why is there a private repository checked out? Is this from a testing branch?

File: antora-playbook.yml [hunk 2]
```diff
@@ -28,7 +28,44 @@ ui:
...
+    allow-uri-read: true
+    # use SVG unless otherwise specified
+    kroki-default-format: svg
+    # Make SVG inline so that we can get interactive (can be overridden by schema with "opts=none")
+    kroki-default-options: interactive
+    sectnums: "@"
+    xrefstyle: short@
+    listing-caption: Code
+    page-width-limit: none@
+    THIS_STANDARD: ASAM PLACEHOLDER
```
Comment (ClemensLinnhoff) [inline antora-playbook.yml:52]: What is this placeholder for?

**Review by ClemensLinnhoff — CHANGES_REQUESTED** [review]

File: .github/workflows/antora-build.yml (hunk 1 above)
Comment (martinfiebig) [reply .github/workflows/antora-build.yml]: Yes, this was a leftover from my local testing branch, I've removed the private repository.

File: antora-playbook.yml (hunk 2 above)
Comment (martinfiebig) [reply antora-playbook.yml]: This placeholder was copied from the asam-antora-extension code - I've replaced it with the value for OpenMATERIAL.

**Review by ClemensLinnhoff — APPROVED** [review]

**Pull request merged**